tests/API/Management/ClientsTest.php

```diff
@@ -4,40 +4,141 @@
+ * Class ClientsTest
+ *
+ * @package Auth0\Tests\API\Management
+ */
+class ClientsTest extends BasicCrudTest
+{
+    /**
+     * Unique identifier name for Clients.
+     *
+     * @var string
+     */
+    protected $id_name = 'client_id';
 
-        $this->domain = $env['DOMAIN'];
+    /**
+     * Random number used for unique testing names.
+     *
+     * @var integer
+     */
+    protected $rand;
 
-        $api = new Management($token, $env['DOMAIN']);
+    /**
+     * ClientsTest constructor.
+     */
+    public function __construct()
+    {
+        parent::__construct();
+        $this->rand = rand();
+    }
 
+    /**
+     * Return the Clients API to test.
+     *
+     * @return Management\Clients
+     */
+    protected function getApiClient()
+    {
+        $token = $this->getToken($this->env, [ 'clients' => [ 'actions' => ['create', 'read', 'delete', 'update' ] ] ]);
+        $api = new Management($token, $this->domain);
         return $api->clients;
     }
 
-    protected function getCreateBody() {
-        $client_name = 'test-create-client' . rand();
-        echo "\n-- Using client name $client_name \n";
+    /**
+     * Get the Client create data to send with the test create call.
+     *
+     * @return array
+     */
+    protected function getCreateBody()
+    {
+        return [
+            'name' => 'TEST-CREATE-CLIENT-' . $this->rand,
+            'app_type' => 'regular_web',
+            'sso' => false,
+            'description' => '__Auth0_PHP_initial_app_description__',
+        ];
+    }
+
+    /**
+     * Tests the \Auth0\SDK\API\Management\Clients::getAll() method.
+     *
+     * @param array $created_entity - Entity created during create() test.
+     *
+     * @return mixed
+     *
+     * @throws \Exception
+     */
+    protected function getAllEntities($created_entity)
+    {
+        $fields = array_keys($this->getCreateBody());
+        $fields[] = $this->id_name;
+        $page_num = 1;
+
+        // Get the second page of Clients with 1 per page (second result).
+        $paged_results = $this->api->getAll($fields, true, $page_num, 1);
+
+        // Make sure we only have one result, as requested.
+        $this->assertEquals(1, count($paged_results));
 
-        return ['name' => $client_name, 'sso' => false];
+        // Make sure we only have the 4 fields we requested.
+        $this->assertEquals(count($fields), count($paged_results[0]));
+
+        // Get many results (needs to include the created result if self::findCreatedItem === true).
+        $many_results_per_page = 50;
+        $many_results = $this->api->getAll($fields, true, 0, $many_results_per_page);
+
+        // Make sure we have at least as many results as we requested.
+        $this->assertLessThan($many_results_per_page, count($many_results));
+
+        // Make sure our paged result above appears in the right place.
+        $this->assertEquals($paged_results[0][$this->id_name], $many_results[$page_num][$this->id_name]);
```

Comment:
at a first glance it looked wrong to use `$page_num` but it's fine. Please add another comment line saying something like "because we previously requested 1 per page, $page_num represents the expected item in the $paged_results"